### PR TITLE
Project design

### PR DESCRIPTION
**Pull request by Konstantsiy**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,16 @@
 # photo-converter
-Service that expose a RESTful API to convert JPEG to PNG and vice versa and compress the image with the compression ratio specified by the user.
+Service that expose a RESTful API to convert JPEG to PNG and vice versa and compress the image 
+with the compression ratio specified by the user. The user has the ability to view
+the history and status of their requests (for example, queued, processed, completed) and upload 
+the original image and the processed one.
+# Architecture Diagram
+![alt text](./docs/architecture-diagram.png)
+# Endpoints
+- /user/requests - view user requests history [GET]
```
Comment (papahmsolo) [inline README.md:9]: We can leave it as is, but logically requests is more a list of conversions, so maybe it should not be under users path

File: README.md [hunk 2]
```diff
@@ -1,2 +1,16 @@
...
+- /user/login - user authorization [POST]
+- /user/signup - user registration [POST]
+- /user/logout - log out of the user [GET]
+- /convert - convert photo [POST]
+- /download/{photoId} - download needed photo [GET]
```
Comment (papahmsolo) [inline README.md:14]: Let's not use verbs in the path, GET is a verb already and we're downloading images

File: scripts/db.sql [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            create type status as enum ('queued', 'processed', 'failed', 'done');
+        end if;
+    end;
+$$;
+
+create table if not exists converter.photos (
+    id serial primary key,
+    name varchar(80) not null,
+    format file_format not null
+);
```
Comment (papahmsolo) [inline scripts/db.sql:29]: Where do we get the info from the table about where to download the image from?

File: scripts/db.sql [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+create table if not exists converter.requests (
+    id serial primary key,
+    user_id int not null,
+    photo_id int not null,
```
Comment (papahmsolo) [inline scripts/db.sql:34]: It's not necessarily a photo, so let's use image instead

File: scripts/db.sql [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+    source_format file_format not null,
+    final_format file_format not null,
```
Comment (papahmsolo) [inline scripts/db.sql:37]: target?

File: scripts/db.sql [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    ratio int not null,
+    status status not null,
+    created timestamp not null,
```
Comment (papahmsolo) [inline scripts/db.sql:40]: we'll also be updating the table, let's store this info too

**Review by papahmsolo — CHANGES_REQUESTED** [review]
Looks good for a start!

File: README.md (hunk 1 above)
Comment (Konstantsiy) [reply README.md:9]: Interesting remark, thanks

File: scripts/create_db.sql [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    name varchar(80) not null,
+    format file_format not null,
+    location text not null
+);
+
+create table if not exists converter.requests (
+    id uuid default uuid_generate_v1() primary key,
+    user_id uuid not null,
+    source_id uuid not null,
+    target_id uuid not null,
```
Comment (papahmsolo) [inline scripts/create_db.sql:50]: Does it always exist?

File: scripts/create_db.sql [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+    email varchar(50) unique not null,
+    password varchar(50) not null
+);
+
+create table if not exists converter.images (
+    id uuid default uuid_generate_v1() primary key,
+    name varchar(80) not null,
+    format file_format not null,
+    location text not null
+);
```
Comment (papahmsolo) [inline scripts/create_db.sql:44]: user and image is also created somewhen, it's a good practice to store this info too

**Review by papahmsolo — CHANGES_REQUESTED** [review]
Almost there

**Review by papahmsolo — APPROVED** [review]
👍

**Pull request merged**